levels/level_manager.go

```diff
@@ -398,10 +399,23 @@ outer:
 				}
 				continue outer
 			}
+			if len(tables) == 0 {
+				continue
+			}
 			if level == 0 {
-				// Level 0 is overlapping
-				for _, table := range tables {
-					overlapping = append(overlapping, []QueryTableInfo{{ID: table.SSTableID, DeadVersions: table.DeadVersionRanges}})
+				tablesByL0Group := lm.groupTablesByProcessorID(tables)
```

Comment:
This doesn't work, as it assembles query results based on a combination of the readable state, and the current writable state, protected by a RW lock. Using a lock also defeats the purpose of readable state which is to avoid locking. If a compaction or registration occurs, it will get a write lock which will block all readers causing queries to stall while that is happening.

Reply:
I think what we need to do is to add everything that we need to create the query results to the readable state.